src/prefect/environments/storage/github.py

```diff
@@ -0,0 +1,131 @@
+from typing import TYPE_CHECKING, Any, Dict, List
+
+
+from prefect.environments.storage import Storage
+from prefect.utilities.storage import extract_flow_from_file
+
+if TYPE_CHECKING:
+    from prefect.core.flow import Flow
+
+
+class GitHub(Storage):
+    """
+    GitHub storage class. This class represents the Storage interface for Flows stored
+    in `.py` files in a GitHub repository.
+
+    This class represents a mapping of flow name to file paths contained in the git repo,
+    meaning that all flow files should be pushed independently. A typical workflow using
+    this storage type might look like the following:
+
+    Compose flow `.py` file where flow has GitHub storage:
+
+    ```python
+    flow = Flow("my-flow")
+    flow.storage = GitHub(repo="my/repo", path="/flows/flow.py")
+    ```
+
+    Push this `flow.py` file to the `my/repo` repository under `/flows/flow.py`.
+
+    Call `prefect register -f flow.py` to register this flow with GitHub storage.
+
+    Args:
+        - repo (str): the name of a GitHub repository to store this Flow
+        - path (str, optional): a path pointing to a flow file in the repo
```

Comment:
The way this is currently implemented is only one flow to a GitHub storage object. All other pickle based storage options allow N flows to a single storage object. This can be changed by updating the `add_flow` call to take in a `path` pointing to the flow file in a repository but then would require that to be provided during registration. The issue here is we don't do the actual committing of the flow code to the repo so it doesn't get to make any assumptions about location of the flow file. i.e. Generating a key in a pickle based storage option is only useful because those storage options are also uploading the pickle to that set location.